GeneratorInterface/Pythia8Interface/src/ResonanceDecayFilterHook.cc

```diff
@@ -72,12 +76,15 @@ bool ResonanceDecayFilterHook::checkVetoResonanceDecays(const Event& process) {
     if ( (did == 2 || did==3 || did==4 || did==5 ) && udscbAsEquivalent_) {
       did = 1;
     } 
+    if( (did == 23 || did == 24) && wzAsEquivalent_) {
+      did = 23;
+    }
     
     int mid = p.mother1()>0 ? std::abs(process[p.mother1()].id()) : 0;
     
     //if no list of mothers is provided, then all particles
     //in hard process and resonance decays are counted together
-    bool validMother = mothers_.size() ? false : true;
+    bool validMother = mothers_.empty();
```

Comment:
are you sure you have not inverted the logic here?

Reply:
(probably not, just make sure)

Reply:
@perozzi yes, I'm sure: !size() = empt().

Reply:
hi @acarvalh - if this line is correct, then the loop just below is not needed. perhaps it should really be
bool validMother=false?[in fact the declaration of validMother can be avoided trivially]